src/main/java/Nebula.java

```diff
@@ -0,0 +1,189 @@
+import java.util.Scanner;
+
+public class Nebula {
+    /**
+     * Entry point of the application. Initializes the UI, task list, and parser,
+     * then processes user commands in a loop until "bye" is entered
+     *
+     * @param args command-line arguments (not used)
+     */
+    public static void main(String[] args) {
+        Ui ui = new Ui();
+        TaskList taskList = new TaskList();
+        Parser parser = new Parser();
+
+        System.out.println(ui.greeting());
+
+        Scanner sc = new Scanner(System.in);
+        while(sc.hasNext()) {
+            String command = sc.nextLine();
+
+            if(command.equals("bye")) {
+                System.out.println(ui.goodbye());
+                break;
+            }
+
+            else if(command.equals("list")) {
+                System.out.println(ui.displayList());
+            }
+
+            else if(command.startsWith("mark")) {
+                try {
+                    validateCommand(command, taskList);
+                } catch (NebulaException e) {
+                    System.out.println(e.getMessage());
+                    continue;
+                }
+                int taskNum = parser.splitCommandAndTaskNumber(command);
+                System.out.println(taskList.markTask(taskNum));
+            }
+
+            else if(command.startsWith("unmark")) {
+                try {
+                    validateCommand(command, taskList);
+                } catch (NebulaException e) {
+                    System.out.println(e.getMessage());
+                    continue;
+                }
+                int taskNum = parser.splitCommandAndTaskNumber(command);
+                System.out.println(taskList.unmarkTask(taskNum));
+            }
+
+            else if(command.startsWith("delete")) {
+                try {
+                    validateCommand(command, taskList);
+                } catch (NebulaException e) {
+                    System.out.println(e.getMessage());
+                    continue;
+                }
+                int taskNum = parser.splitCommandAndTaskNumber(command);
+                System.out.println(taskList.deleteTask(taskNum));
+            }
+
+            else {
+                try {
+                    validateCommand(command, taskList);
+                } catch (NebulaException e) {
+                    System.out.println(e.getMessage());
+                    continue;
+                }
+
+                TaskType taskType = parseTaskType(command);
+
+                switch (taskType) {
```

Comment:
Good us of switch and case. But the indentation level of switch and case should be the same according to the coding standard that we adhere to.